core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -79,4 +87,19 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Spar
   override def onUnpersistRDD(unpersistRDD: SparkListenerUnpersistRDD) = synchronized {
     _rddInfoMap.remove(unpersistRDD.rddId)
   }
+
+  override def onBlockUpdate(blockUpdateEvent: SparkListenerBlockUpdate) = synchronized {
+    // only update broadcast for now, need to be modified if want to track other blocks
+    val broadcastIdOpt = blockUpdateEvent.blockId.asBroadcastId
+    if (broadcastIdOpt.isDefined) {
```

Comment:
I don't think it particularly matters here, but fyi a more "scala" way of doing this is:

```
blockUpdateEvent.blockId.asBroadcastId.foreach{broadcastBlock =>
  val broadcastId = broadcastBlock.broadcastId
  ...
}
```